Parameterize prunes in check_format

ddn0 opened the pull request:
Also add cu as a possible file extension. I didn't parameterize file
extensions for check_format (yet) because the set of viable extensions
for clang-format is small and probably well-known to check_format.sh
while the set of paths to exclude is context dependent.

witchel commented on libtsuba/include/tsuba/tsuba.h:
```diff
@@ -1,29 +1,71 @@
+ * @begin and @size should be well aligned to TSUBA_BLOCK_SIZE
+ * return value will be aligned to tsuba_block_size as well
+ */
+uint8_t* Mmap(const std::string& filename, uint64_t begin, uint64_t size);
```
Probably too detailed, but why is this called Mmap?  Do we really want to support partial downloads?  Do we have to remember that we only downloaded part of this file when we try to write it back?

In general, I'm getting the vibe that the storage layer should be in charge of memory management when it relates to buffering files.  That isn't so compatable with an mmap-like interface.

ddn0 replied:
FWIW, my view is that in the limit we probably have to bubble up buffering to the application layer (like how MPI has blocking and nonblocking sends), but in the short-term, most applications will be a single bulk-load, compute and then single bulk-store.

How this shakes out in terms of an interface and its evolution is another story.

tylershunt replied:
Yeah Mmap isn't the right name here, but I expect the api to evolve a lot so I'm going to hold off on optimizing it.

ddn0 commented on libgalois/src/FileView.cpp:
```diff
@@ -15,36 +15,36 @@ FileView::~FileView() { Unbind(); }
-  if (err) {
+  tsuba::StatBuf buf;
+  int err = tsuba::Stat(filename, &buf);
+  if (!err) {
```
`if (err)`; this is already fixed in master.

tylershunt replied:
Yeah sloppy merge my bad.

ddn0 committed:
Parameterize prunes in check_format

Also add cu as a possible file extension. I didn't parameterize file
extensions for check_format (yet) because the set of viable extensions
for clang-format is small and probably well-known to check_format.sh
while the set of paths to exclude is context dependent.

arthurp reviewed (approved):
LGTM.

roshandathathri commented on scripts/check_format.sh:
```diff
@@ -34,7 +34,7 @@ while read -d '' filename; do
     fi
   fi
-done < <(find ${ROOTS} $(emit_prunes) -name '*.cpp' -print0 -o -name '*.h' -print0)
+done < <(find ${ROOTS} $(emit_prunes) -name '*.cpp' -print0 -o -name '*.h' -print0 -o -name '*.cu' -print0)
```
Sorry, my bad. I should have mentioned that we also have *.cuh files in our repo. Can you please add that too?

roshandathathri reviewed (approved):
LGTM
